### PR TITLE
admin ui styling, edit/add/del categories

### PR DESCRIPTION
**Pull request by maggiemay-afk**
Updated Admin UI styling to be dark mode
Updated styling and form control for video editor form
Added editor form for categories
(Edit, Add, Delete categories)
Added snack bar for success/error messages when attempting to make changes to the DB

### CODE REVIEW

File: src/admin.tsx [hunk 1]
```diff
@@ -171,9 +291,19 @@ export default function AdminPage() {
           aria-label="file system navigator"
           defaultCollapseIcon={<ExpandMoreIcon />}
           defaultExpandIcon={<ChevronRightIcon />}
-          sx={{ height: '100%', flexGrow: 1, maxWidth: 600, overflowY: 'auto', paddingLeft: 10 }}
+          sx={{ height: '100%', flexGrow: 1, maxWidth: 900, overflowY: 'auto', paddingLeft: 10 }}
```
Comment (maggiemay-afk) [inline src/admin.tsx:294]: bumped the width to allow more space for input boxes in the nested tree items

File: src/admin.tsx [hunk 2]
```diff
@@ -139,29 +225,57 @@ export default function AdminPage() {
...
+          >
             {vids.map(vid => (
               <TreeItem
                 expandIcon={<TreeIcon tooltipText='Edit Video' icon={<BuildCircleIcon />} />}
-                collapseIcon={<TreeIcon tooltipText='Expand' icon={<BuildCircleIcon color='error' />} />}
-                key={vid.name}
-                nodeId={vid.name}
+                collapseIcon={<TreeIcon tooltipText='Close Editor' icon={<BuildCircleIcon color='error' />} />}
+                key={documentId + subcategory + vid.code + "edit_video"}
+                nodeId={documentId + subcategory + vid.code + "edit_video"}
```
Comment (maggiemay-afk) [inline src/admin.tsx:255]: Updating keys and nodeIds throughout the tree to ensure that each one is truly unique

File: src/components/VideoEditor.tsx [hunk 3]
```diff
@@ -132,123 +133,143 @@ export default function VideoEditor(props: VideoEditorProps) {
...
+          variant="outlined"
           size='small'
+          color='secondary'
           select
-          value={localVideo?.livestream}
+          value={!!localVideo?.livestream}
         >
-          <MenuItem key="true" value="true">Yes</MenuItem>
-          <MenuItem key="false" value="false">No</MenuItem>
+          <MenuItem key="true" value="true">True</MenuItem>
```
Comment (maggiemay-afk) [inline src/components/VideoEditor.tsx:200]: For maximum clarity, using only true and false in the Menu Items

File: src/error-page.tsx
Comment (maggiemay-afk) [inline src/error-page.tsx:1]: Page was originally created when adding react router. Would have been the custom error page that displayed, deemed not necessary for the time being

File: src/index.css [hunk 4]
```diff
@@ -35,26 +35,13 @@ code {
   padding-bottom: calc(30px + env(safe-area-inset-bottom));
 }
 
-/* body {
```
Comment (maggiemay-afk) [inline src/index.css:38]: This code was commented out already, decided to delete

File: src/App.tsx [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
+import { theme, adminTheme } from './theme';
 import { RecoilRoot } from 'recoil';
-import { Route, Routes } from "react-router-dom";
+import { Route, Routes, useLocation } from "react-router-dom";
 import LoginPage from './login';
 import AdminPage from './admin';
 import { RequireAuth } from './RequireAuth';
 import { AuthProvider } from './AuthProvider';
-
+import CssBaseline from '@mui/material/CssBaseline';
```
Comment (maggiemay-afk) [inline src/App.tsx:11]: Needed for MUI's standard dark mode theme

Comment (maggiemay-afk) [inline src/App.tsx:11]: Needed for MUI's standard dark mode theme

**Comment by maggiemay-afk** [comment]
Quite a bit of code cleanup here as well:
(For some of these I also left comments on the file as well) 
- remove unused error page
- remove commented out code in CSS file
- remove unused styling in CSS file
- rename some components and functions for clarity 
- reorganize imports to help with all the MUI components 
- update tree structure nodeIds and keys to ensure they're all unique 
- other reorganizing and restructuring of code

File: src/App.tsx [hunk 6]
```diff
@@ -1,21 +1,25 @@
...
 
 const App: React.FC = () => {
 
+  const adminThemeMap = ["/login", "/admin"];
```
Comment (hermanator608) [inline src/App.tsx:15]: This variable name is a bit confusing, should this be a map (Or maybe a Set)? Would make the check below instant time lookup

File: src/admin.tsx [hunk 7]
```diff
@@ -69,10 +109,12 @@ export default function AdminPage() {
...
     newVideos[videoIndex] = newData;
 
-
     const docRef = doc(db, AMBIANCE_COLLECTION, documentId);
     return updateDoc(docRef, {
       videos: newVideos
+    }).then(() => {
+      setAlertSeverity("success");
+      handleAddSnackBarMessage("Successfully updated video - " + newData.name);
     });
```
Comment (hermanator608) [inline src/admin.tsx:118]: Do we want to add error handling to all of these in this PR? If so lets add an Error SnackBar message

File: src/admin.tsx [hunk 8]
```diff
@@ -171,24 +285,46 @@ export default function AdminPage() {
...
   }
 
-
   return (
     <div id="admin-page" data-testid='admin'>
-      <Box sx={{ flexGrow: 1 }}>
-        <AppBar className="app-bar" >
+      <Snackbar
+        open={open}
+        autoHideDuration={6000}
```
Comment (hermanator608) [inline src/admin.tsx:310]: Lets pull out this constant to a variable at the top of this file

File: src/login.tsx [hunk 9]
```diff
@@ -57,13 +58,13 @@ export default function LoginPage() {
       id="login-page"
       data-testid='login'
       component="form"
+      noValidate
```
Comment (hermanator608) [inline src/login.tsx:61]: What is this for?

File: src/login.tsx [hunk 10]
```diff
@@ -57,13 +58,13 @@ export default function LoginPage() {
...
+      autoComplete="off"
```
Comment (hermanator608) [inline src/login.tsx:62]: Should we enable autocomplete for login?

**Review by hermanator608 — CHANGES_REQUESTED** [review]
Great work! Love how easy it was to add on to the tree with new category editing!

Few small questions/suggestions

File: src/admin.tsx (hunk 7 above)
Comment (maggiemay-afk) [reply src/admin.tsx:118]: Done. I tested the error handling by temporarily changing Firestore rules to "false", not allowing any actions on the database.

File: src/login.tsx (hunk 9 above)
Comment (maggiemay-afk) [reply src/login.tsx:61]: I can see these fields being used in some MUI examples, in their docs, which is why I had them. But I don't see any difference in behavior, so I removed the noValidate and I also changed it to autoComplete="on". Autocomplete isn't perfect, but it's a nice-to-have.

File: src/admin.tsx (hunk 8 above)
Comment (maggiemay-afk) [reply src/admin.tsx:310]: Can we use the constants.ts file? That's where I put it for now, but let me know if you disagree!

File: src/login.tsx (hunk 10 above)
Comment (maggiemay-afk) [reply src/login.tsx:62]: Changed this before I saw this comment. See my response above!

**Comment by maggiemay-afk** [comment]
Also adding to this PR:

- Expand / Collapse all nodes functionality for the tree

File: src/admin.tsx [hunk 11]
```diff
@@ -54,6 +56,66 @@ export default function AdminPage() {
...
+
+    const videosBySubCat = groupVideosBySubcategory(data);
+    const expandedItems: string[] = [];
+
+    Object.entries(videosBySubCat).forEach(([documentID, subcategoryGrouping]) => {
+      expandedItems.push(documentID);
+      Object.keys(subcategoryGrouping.videosBySubcategory).forEach((subcategory) => {
+        expandedItems.push(documentID + subcategory);
+      })
+    })
```
Comment (hermanator608) [inline src/admin.tsx:110]: So this list is generated _every_ time we press the button. Can we find a way to only generate this once? Maybe a new state variable which we populate in the existing useEffect after we get data from firebase?

**Review by hermanator608 — CHANGES_REQUESTED** [review]
Great work - one suggestion

**Review by hermanator608 — APPROVED** [review]
LGTM - Great work 💯

**Pull request merged**